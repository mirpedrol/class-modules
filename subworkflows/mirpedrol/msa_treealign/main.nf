include { FAMSA_TREEALIGN } from '../../../modules/mirpedrol/famsa/treealign/main'
include { MAGUS_TREEALIGN } from '../../../modules/mirpedrol/magus/treealign/main'
include { CLUSTALO_TREEALIGN } from '../../../modules/mirpedrol/clustalo/treealign/main'
include { TCOFFEE_TREEALIGN } from '../../../modules/mirpedrol/tcoffee/treealign/main'


workflow MSA_TREEALIGN {

    take:
    ch_fasta
    ch_tree

    main:
    def ch_out_alignment = Channel.empty()
    def ch_out_versions = Channel.empty()

    ch_fasta
        .branch {
            meta, fasta, tool ->
                famsa_treealign: tool == "famsa_treealign"
                    return [ meta, fasta ]
                magus_treealign: tool == "magus_treealign"
                    return [ meta, fasta ]
                clustalo_treealign: tool == "clustalo_treealign"
                    return [ meta, fasta ]
                tcoffee_treealign: tool == "tcoffee_treealign"
                    return [ meta, fasta ]
        }
        .set { ch_fasta_branch }
    ch_tree
        .branch {
            meta, tree, tool ->
                famsa_treealign: tool == "famsa_treealign"
                    return [ meta, tree ]
                magus_treealign: tool == "magus_treealign"
                    return [ meta, tree ]
                clustalo_treealign: tool == "clustalo_treealign"
                    return [ meta, tree ]
                tcoffee_treealign: tool == "tcoffee_treealign"
                    return [ meta, tree ]
        }
        .set { ch_tree_branch }

    FAMSA_TREEALIGN( ch_fasta_branch.famsa_treealign, ch_tree_branch.famsa_treealign )
    ch_out_alignment = ch_out_alignment.mix(FAMSA_TREEALIGN.out.alignment)
    ch_out_versions = ch_out_versions.mix(FAMSA_TREEALIGN.out.versions)

    MAGUS_TREEALIGN( ch_fasta_branch.magus_treealign, ch_tree_branch.magus_treealign )
    ch_out_alignment = ch_out_alignment.mix(MAGUS_TREEALIGN.out.alignment)
    ch_out_versions = ch_out_versions.mix(MAGUS_TREEALIGN.out.versions)

    CLUSTALO_TREEALIGN( ch_fasta_branch.clustalo_treealign, ch_tree_branch.clustalo_treealign )
    ch_out_alignment = ch_out_alignment.mix(CLUSTALO_TREEALIGN.out.alignment)
    ch_out_versions = ch_out_versions.mix(CLUSTALO_TREEALIGN.out.versions)

    TCOFFEE_TREEALIGN( ch_fasta_branch.tcoffee_treealign, ch_tree_branch.tcoffee_treealign )
    ch_out_alignment = ch_out_alignment.mix(TCOFFEE_TREEALIGN.out.alignment)
    ch_out_versions = ch_out_versions.mix(TCOFFEE_TREEALIGN.out.versions)



    emit:
    alignment = ch_out_alignment
    versions = ch_out_versions

}

