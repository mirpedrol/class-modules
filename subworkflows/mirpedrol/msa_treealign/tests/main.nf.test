nextflow_workflow {

    name "Test Subworkflow MSA_TREEALIGN"
    script "../main.nf"
    workflow "MSA_TREEALIGN"

    tag "subworkflows"
    tag "subworkflows_mirpedrol"
    tag "subworkflows/msa_treealign"
    tag "famsa/treealign"
    tag "magus/treealign"
    tag "clustalo/treealign"
    tag "tcoffee/treealign"
    tag "famsa/guidetree"
    tag "magus/guidetree"
    tag "clustalo/guidetree"


    test("run famsa/treealign") {

        setup {
            run("FAMSA_GUIDETREE") {
                script "../../../../modules/mirpedrol/famsa/guidetree/main.nf"
                process {
                    """
                    input[0] = [ [ id:'test' ], // meta map
                                    file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                                ]
                    """
                }
            }
        }

        when {
            params.treealign = "famsa/treealign"
            workflow {
                """
                input[0] = [ [ id:'test' ], // meta map
                                file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                            ]
                input[1] = FAMSA_GUIDETREE.out.tree.collect{ meta, tree -> tree }.map{ tree -> [[ id: 'test_summary'], tree]}
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.alignment).match("famsa_treealign_alignment")},
                { assert snapshot(workflow.out.versions).match("famsa_treealign_versions") }
            )
        }
    }

    test("run magus/treealign") {

        setup {
            run("MAGUS_GUIDETREE") {
                script "../../../../modules/mirpedrol/magus/guidetree/main.nf"
                process {
                    """
                    input[0] = [ [ id:'test' ], // meta map
                                    file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                                ]
                    """
                }
            }
        }

        when {
            params.treealign = "magus/treealign"
            workflow {
                """
                input[0] = [ [ id:'test' ], // meta map
                                file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                            ]
                input[1] = MAGUS_GUIDETREE.out.tree.collect{ meta, tree -> tree }.map{ tree -> [[ id: 'test_summary'], tree]}
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                // tests seem to be reproducible on a single machine, but not across different machines
                // test the correct samples are in there
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1apf") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1ahl") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1atx") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1sh1") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1bds") },
                { assert snapshot(workflow.out.versions).match("magus_treealign_versions") }
            )
        }
    }

    test("run clustalo/treealign") {

        setup {

            run("CLUSTALO_GUIDETREE") {
                script "../../../../modules/mirpedrol/clustalo/guidetree/main.nf"
                process {
                    """
                    input[0] = [ [ id:'test' ], // meta map
                                    file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                                ]
                    """
                }
            }
        }

        when {
            params.treealign = "clustalo/treealign"
            workflow {
                """
                input[0] = [ [ id:'test' ], // meta map
                                file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                            ]
                input[1] = CLUSTALO_GUIDETREE.out.tree.collect{ meta, tree -> tree }.map{ tree -> [[ id: 'test_summary'], tree]}
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.alignment).match("clustalo_treealign_alignment")},
                { assert snapshot(workflow.out.versions).match("clustalo_treealign_versions") }
            )
        }
    }

    test("run tcoffee/treealign") {

        setup {

            run("FAMSA_GUIDETREE") {
                script "../../../../modules/mirpedrol/famsa/guidetree/main.nf"
                process {
                    """
                    input[0] = [ [ id:'test' ],
                                    file(params.test_data['sarscov2']['genome']['informative_sites_fas'], checkIfExists: true)
                                ]

                    """
                }
            }
        }

        when {
            params.treealign = "tcoffee/treealign"
            workflow {
                """
                input[0] = [ [ id:'test' ],
                                file(params.test_data['sarscov2']['genome']['informative_sites_fas'], checkIfExists: true)
                            ]
                input[1] = FAMSA_GUIDETREE.out.tree.collect{ meta, tree -> tree }.map{ tree -> [[ id: 'test'], tree]}
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.alignment).match("tcoffee_treealign_alignment")},
                { assert snapshot(workflow.out.versions).match("tcoffee_treealign_versions") }
            )
        }
    }
}