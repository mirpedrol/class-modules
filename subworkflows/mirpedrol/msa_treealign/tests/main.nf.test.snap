{
    "clustalo_treealign_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,74bb9a2820a91cf68db94dbd46787722"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-22T16:01:01.611023951"
    },
    "magus_treealign_versions": {
        "content": [
            [
                "versions.yml:md5,32409a4801ce34648c31cef62c4bb298"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-22T16:00:05.747842102"
    },
    "clustalo_treealign_versions": {
        "content": [
            [
                "versions.yml:md5,7d9dc598369d063294d8ff5b62d02e2f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-22T16:01:01.661726627"
    },
    "magus_versions": {
        "content": [
            [
                "versions.yml:md5,6bd0fc2decf3dd8e3b43a4d5b9c0cc58"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T15:21:31.000053"
    },
    "tcoffee_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,93bc8adfcd88f7913718eacc13da8e4a"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T15:21:57.317953"
    },
    "tcoffee_treealign_versions": {
        "content": [
            [
                "versions.yml:md5,c9000e613bf71e55f37a77eacd606fcf"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-22T16:01:18.493259432"
    },
    "famsa_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,7cf7375f2ba360814ea978731838b972"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T15:21:17.914586"
    },
    "famsa_versions": {
        "content": [
            [
                "versions.yml:md5,3002c8ce2c4f1b9a4b084b61efc0c2b4"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T15:21:17.922286"
    },
    "clustalo_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,74bb9a2820a91cf68db94dbd46787722"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T15:21:52.483352"
    },
    "clustalo_versions": {
        "content": [
            [
                "versions.yml:md5,2c9f1ab1c3e8a431546fda2ef0170713"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T15:21:52.490603"
    },
    "tcoffee_versions": {
        "content": [
            [
                "versions.yml:md5,c6bb7f0fcab261972bd722de048f36b1"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T15:21:57.324541"
    },
    "famsa_treealign_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,7cf7375f2ba360814ea978731838b972"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-22T15:59:47.187845271"
    },
    "tcoffee_treealign_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,93bc8adfcd88f7913718eacc13da8e4a"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-22T16:01:18.433917307"
    },
    "famsa_treealign_versions": {
        "content": [
            [
                "versions.yml:md5,d62584b93ac23e3e3089a9c50f48e4e8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-22T15:59:47.275076014"
    }
}